util/gem5-resources-manager-v2/src/gem5_resources_manager/creator/creator.py

```diff
@@ -0,0 +1,154 @@
+        self.data_source = data_source
+        print("Creator.execute()", args.category)
+        self.create_resource_json(args)
+
+    def create_resource_json(self, args: Namespace):
+        resource = {}
+        required, optional = self.data_source.get_fields(args.category)
+        populated_fields = ast.literal_eval(args.field_entries)
+        resource["category"] = args.category
+        del required["category"]
+
+        self.enter_fields(required, resource, populated_fields, args)
+        if args.required_fields_only:
+            self.enter_fields(
+                optional, resource, populated_fields, args, is_optional=True
+            )
+
+        if not args.ignore_db_check and self.data_source.resource_exists(
+            resource_id=resource["id"]
+        ):
+            # throw runtime exception
+            raise Exception("Resource already exists in database")
+
+        if not args.ignore_schema_validation:
+            self.data_source.validate_resource(resource)
+        self.data_source.create_new_entry(resource)
+
+    def enter_fields(self, fields, resource, populated_fields, args, is_optional=False):
+        for field in fields.keys():
+            if field in populated_fields:
+                resource[field] = populated_fields[field]
+                print(f"{field}: {resource[field]}")
+                if field == "url" and args.handle_url:
+                    self.handle_url(resource[field], populated_fields)
+                continue
+            if is_optional:
+                choice = input(
+                    f"'{field}' is an optional field. Enter Y to enter a value for this field, or N to skip this field."
+                )
+                if choice.lower() == "n":
+                    continue
+                elif choice.lower() != "y":
+                    raise Exception("Invalid input")
+            if args.verbose:
+                print(
+                    f"The {field} takes the following input: {json.dumps(fields[field], indent=4)}"
+                )
+            user_input = input(f"{field}: ")
+            if (
+                fields[field]["type"] == "integer"
+                or fields[field]["type"] == "array"
+                or fields[field]["type"] == "object"
+            ):
+                resource[field] = ast.literal_eval(user_input)
+            else:
+                resource[field] = user_input
+            if field == "url" and args.handle_url:
+                self.handle_url(resource[field], populated_fields)
+
+    def handle_url(self, url, populated_fields):
+        urllib.request.urlretrieve(url, "temp")
+        is_zipped = zipfile.is_zipfile("temp")
+        is_tarred = tarfile.is_tarfile("temp")
+        # if url ends with .tar or .tar.gz, then it is tarred
+        if url.endswith(".tar"):
+            is_tarred = True
+        if url.endswith(".tar.gz"):
+            is_tarred = True
+            is_zipped = True
```

Comment:
Don't do it this way.  You can't infer file type by extension. Use Python's tar and gzip library